django/test/client.py

```diff
@@ -592,39 +592,48 @@ def login(self, **credentials):
         are incorrect, or the user is inactive, or if the sessions framework is
         not available.
         """
-        from django.contrib.auth import authenticate, login
+        from django.contrib.auth import authenticate
         user = authenticate(**credentials)
         if (user and user.is_active and
                 apps.is_installed('django.contrib.sessions')):
-            engine = import_module(settings.SESSION_ENGINE)
+            self._login(user)
+            return True
+        else:
+            return False
 
-            # Create a fake request to store login details.
-            request = HttpRequest()
+    def force_login(self, user, backend=None):
```

Comment:
I'd actually prefer something like `force_login` for exactly that reason. `login_user` suggests something more generic than shortcircuting the `is_active` flag.

On June 16, 2015 10:10:32 PM GMT+02:00, Marc Tamlyn <notifications@github.com> wrote:
>>  
>> -            # Create a fake request to store login details.
>> -            request = HttpRequest()
>> +    def force_login(self, user, backend=None):
>
>Not a bad plan at all. I originally suggested `force_login` on the
>ticket to make it clear that it bypasses the authentication systems
>completely. Everything here is pretty clearly a testing utility mind.
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/django/django/pull/4865/files#r32564941